### ISSUE TITLE
PushNotification undefined (line 6225)

### ISSUE THREAD

**Issue by benlooi** [phonegap/phonegap-plugin-push#1185]
### Expected Behaviour
I am having problems with the plugin as it throws a white screen of death on my emulator when I inlcude PushNotification in my app. An error was reported that PushNotification is undefined on line 6225.
### Actual Behaviour
Here is line 6225 generated from my Android project folder ( I am using Ionic)

angular.module('ngCordova.plugins.push_v5', [])
  .factory('$cordovaPushV5',['$q', '$rootScope', '$timeout', function ($q, $rootScope, $timeout) {
   /*global PushNotification*/

    var push;
    return {
      initialize : function (options) {
        var q = $q.defer();
    //line 6225 //
        push = PushNotification.init(options);
        q.resolve(push);
        return q.promise;
      },

So I tried to change it following the previous old PushNotification with $windows.plugins.PushNotification but when I do cordova emulate android, it overwrites this can puts back push=PushNotification.init, which obviously is not declared as a dependency.
#### Cordova CLI version and cordova platform version

    cordova --version     6.3.1                               # e.g. 6.0.0
    cordova platform version android       5.2.2             # e.g. 4.1.1

#### Plugin version
Plugin version 1.8.1
    cordova plugin version | grep phonegap-plugin-push   # e.g. 1.5.3

**Comment by benlooi** [comment]
Update: further reading and understanding of problem is that PushNotification is supposed to have been defined in push.js.
So for some reason maybe push.js is not loaded.

**Comment by benlooi** [comment]
Ok, tried to compile it on another machine. Works. So I guess it has to do with my Android Studio settings on different machines.  Sorry for the distraction. Will figure out what went wrong with my other machine's setup.

**Comment by benlooi** [comment]
Ok, for those who encounter this problem, I finally realise what went wrong. NOTHING to do with Android or Cordova or this plugin (Great job, by the way.)
I did a blind copy of the example in the ngCordova docs WITHOUT understanding.
     
 module.run(function($http, $cordovaPushV5) {
  // rest of sample code is here 
 }

It ran on my other machine because it is a freakin' fast MacPro, but doesn't run on my iMac or Redmi Note 3.

I then realise the PushNotification is undefined because it hasn't loaded yet. I forgot to wait for device to be ready. So, in my Ionic app.js, I added to the line:
.run(function ($ionicPlatform){
});

So, it ended up like this:  
.run(function ($ionicPlatform, $http, $rootScope,$cordovaPushV5, UserService) {
 //code for windows plugin keyboard stuff
//define the options here
var options = {
    android: {
      senderID: "295998469081"
    },
    ios: {
      alert: "true",
      badge: "true",
      sound: "true"
    },
    windows: {}
  };

 if(window.cordova && $cordovaPushV5) {
   //put the sample code here!
}
}

Key principle : Wait for device to be ready before you call PushNotification.init.
